### PR TITLE
py-bokeh: new version 3.3.1, and supporting packages

### PR DESCRIPTION
**Pull request by wdconinc**
This adds the first bokeh v3 release in spack (a year after the first bokeh v3 release). Some extra dependencies added.

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
FYI. Confirmed the new homepage and both packages' version sha256.

File: var/spack/repos/builtin/packages/py-xyzservices/package.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyXyzservices(PythonPackage):
+    """xyzservices is a lightweight library providing a repository of
+    available XYZ services offering raster basemap tiles."""
+
+    homepage = "https://xyzservices.readthedocs.io/"
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-xyzservices/package.py]:
```suggestion
    homepage = "https://github.com/geopandas/xyzservices"
```
We try to be consistent with what is specified on PyPI.

File: var/spack/repos/builtin/packages/py-bokeh/package.py [hunk 2]
```diff
@@ -12,6 +12,7 @@ class PyBokeh(PythonPackage):
     homepage = "https://github.com/bokeh/bokeh"
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-bokeh/package.py]:
```suggestion
    homepage = "https://bokeh.org/"
```
We try to be consistent with what is specified on PyPI.

File: var/spack/repos/builtin/packages/py-bokeh/package.py [hunk 3]
```diff
@@ -33,11 +38,15 @@ class PyBokeh(PythonPackage):
     depends_on("py-jinja2@2.7:", type=("build", "run"))
     depends_on("py-jinja2@2.9:", type=("build", "run"), when="@2.3.3:")
 
+    depends_on("py-contourpy@1", type=("build", "run"), when="@3:")
+
     depends_on("py-numpy@1.7.1:", type=("build", "run"))
     depends_on("py-numpy@1.11.3:", type=("build", "run"), when="@2.3.3:")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-bokeh/package.py:45]:
```suggestion
    depends_on("py-numpy@1.11.3:", type=("build", "run"), when="@2.3.3:")
    depends_on("py-numpy@1.16:", type=("build", "run"), when="@3.1:")
```

**Review by manuelakuhn — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-bokeh/package.py [hunk 4]
```diff
@@ -20,11 +21,15 @@ class PyBokeh(PythonPackage):
...
     depends_on("py-setuptools", type="build", when="@1.3.4:")
+    depends_on("py-setuptools@64:", type="build", when="@3:")
+    depends_on("py-setuptools-git-versioning", type="build", when="@3:")
+    depends_on("py-colorama", type="build", when="@3:")
 
     depends_on("python@2.6:", type=("build", "run"), when="@0.12.2")
     depends_on("python@2.7:", type=("build", "run"), when="@1.3.4:")
     depends_on("python@3.6:", type=("build", "run"), when="@2.3.3:")
     depends_on("python@3.7:", type=("build", "run"), when="@2.4.0:")
+    depends_on("python@3.8:", type=("build", "run"), when="@3.0.0:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bokeh/package.py:32]: It's 3.9 now

File: var/spack/repos/builtin/packages/py-bokeh/package.py [hunk 5]
```diff
@@ -33,11 +38,16 @@ class PyBokeh(PythonPackage):
     depends_on("py-jinja2@2.7:", type=("build", "run"))
     depends_on("py-jinja2@2.9:", type=("build", "run"), when="@2.3.3:")
 
+    depends_on("py-contourpy@1", type=("build", "run"), when="@3:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bokeh/package.py]:
```suggestion
    depends_on("py-contourpy@1:", type=("build", "run"), when="@3:")
```

File: var/spack/repos/builtin/packages/py-bokeh/package.py [hunk 6]
```diff
@@ -48,3 +58,5 @@ class PyBokeh(PythonPackage):
 
     depends_on("py-typing-extensions@3.7.4:", type=("build", "run"), when="@2.3.3:")
     depends_on("py-typing-extensions@3.10.0:", type=("build", "run"), when="@2.4.0:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bokeh/package.py]: typing-extensions is no longer required

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-bokeh/package.py (hunk 4 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/py-bokeh/package.py:32]:
```suggestion
    depends_on("python@3.8:", type=("build", "run"), when="@3.0.0:")
    depends_on("python@3.9:", type=("build", "run"), when="@3.2.0:")
```

File: var/spack/repos/builtin/packages/py-bokeh/package.py (hunk 6 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/py-bokeh/package.py]:
```suggestion
    depends_on("py-typing-extensions@3.7.4:", type=("build", "run"), when="@2.3.3:3.0.0")
    depends_on("py-typing-extensions@3.10.0:", type=("build", "run"), when="@2.4.0:3.0.0")
```

Comment (wdconinc) [reply var/spack/repos/builtin/packages/py-bokeh/package.py]: ^ Note: yes, 3.0.0 inclusive. Typing extension dependency removed in 3.0.1.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**